lxd/instance/drivers/driver_qemu.go

```diff
@@ -2185,7 +2185,16 @@ func (d *qemu) deviceAttachPath(deviceName string, configCopy map[string]string,
 		return fmt.Errorf("Failed to connect to QMP monitor: %w", err)
 	}
 
-	addr, err := net.ResolveUnixAddr("unix", virtiofsdSockPath)
+	virtiofsdSocketDir, err := os.Open(filepath.Dir(virtiofsdSockPath))
```

Comment:
Can you also use `shortenedSocketPath()` here instead?

Reply:
It is almost the same process, but in this case the file descriptor I am using is from the lxd process itself, `shortenedSocketPath` includes the logic of using a file descriptor so that other process (qemu or swtpm) has access to that file.
In short, using it here would result in probably using the wrong `fd` in `/proc/self/fd`, pointing to some other random file (or no file at all) and not `virtiofsdSocketDir`.

Reply:
That said, I could modify `shortenedSocketPath` to move this part of the logic outside the function or maybe using a flag to change its behavior in this case, either way I don't see how to do this cleanly. Feel free to make any sugggestions!